### ISSUE TITLE
Some weird behaviour of findPattern

### ISSUE THREAD

**Issue by albmar** [Rob--/memoryjs#38]
Once I had the following situation. I ran a code where 2 pattern searches in normal mode are done. The first was correctly done, but the resulting address of the second search gave me just the base address of the module.

My code looks like this:
```javascript
let pattern = Buffer.from('some string value').toString('hex').replace(/..\B/g, '$& ');
let address = memoryjs.findPattern(processObject.handle, module.szModule, pattern, memoryjs.NORMAL, 0, 0);
pattern = Buffer.alloc(4);
pattern.writeUInt32LE(address);
pattern = pattern.toString('hex').replace(/..\B/g, '$& ');
// doesn't work with memoryjs version > 3.0.2
address = memoryjs.findPattern(processObject.handle, module.szModule, pattern, memoryjs.NORMAL, 0, 0);
```

After some testing I found out, that for some reason I cannot explain yet, using a variable for the pattern string leads to the problem. The changes to lib/memoryjs.cc were made after version 3.0.2.

So this will have problems...
```cpp
const char* pattern = std::string(*signature).c_str();
...
address = Pattern.findPattern(handle, moduleEntries[i], pattern, sigType, patternOffset, addressOffset);
```
...and this will work fine.
```cpp
address = Pattern.findPattern(handle, moduleEntries[i], std::string(*signature).c_str(), sigType, patternOffset, addressOffset);
```

**Comment by Rob--** [comment]
Might have to do with passing by reference, just for the sake of ease in the next update I'll revert back to directly passing the pattern.

**Comment by Rob--** [comment]
Addressed in 491d62dac7e854e825e3eae40bb777da750309c1. Thanks for letting me know!
